### PR TITLE
Api/chat

### PR DESCRIPTION
**Pull request by DipanwitaDey20**
In this PR there are add project, update project, cancel project and paused project.

### CODE REVIEW

File: config/mimes.php
Comment (SankalanSaha) [inline config/mimes.php:1]: application/zip is not a docx file
Also add the valid mimetypes for the other file extensions

File: app/Helpers/helpers.php
Comment (SankalanSaha) [inline app/Helpers/helpers.php:1]: Line 337: In the cancel project notification, the first estimate will get paused, however the estimates which is accepted by the customer should be paused.

File: app/Http/Controllers/Api/ProjectController.php [hunk 1]
```diff
@@ -29,13 +33,17 @@ public function show(Request $request, Project $project){
...
             'forename' => 'required|string',
             'surname' => 'required|string',
             'project_name' => 'required|string',
             'description' => 'required|string',
             'contact_mobile_no' => 'required|string',
             'contact_email' => 'required|string',
+            'images' => 'required|array',
+            'images.*' => 'required|max:'.(config('const.dropzone_max_file_size')*1024).'|mimetypes:'.config('const.dropzone_accepted_image'),
+        ],$messages = [
+            'mimes' => 'Please insert image only',
```
Comment (SankalanSaha) [inline app/Http/Controllers/Api/ProjectController.php:45]: Show the supported images in the mimes after fetching from the  const.dropzone_accepted_image

File: app/Http/Controllers/Api/ProjectController.php [hunk 2]
```diff
@@ -64,68 +106,145 @@ public function add_project(Request $request){
...
+    public function paused_project(Request $request){
         $validator = Validator::make($request->all(), [
-            'project_id' => 'required|Project',
-            'file_name' => 'required',
-            'file_extention' => 'required',
-            'url' => 'required',
+            'project_id' => 'required',
+            'status' => 'required'
         ]);
-        if ($validator->fails()) {
```
Comment (SankalanSaha) [inline app/Http/Controllers/Api/ProjectController.php:115]: Why did you remove the validation check?

File: app/Http/Controllers/Api/ProjectController.php [hunk 3]
```diff
@@ -64,68 +106,145 @@ public function add_project(Request $request){
...
-    public function add_file(Request $request){
+
+    public function paused_project(Request $request){
         $validator = Validator::make($request->all(), [
-            'project_id' => 'required|Project',
-            'file_name' => 'required',
-            'file_extention' => 'required',
-            'url' => 'required',
+            'project_id' => 'required',
+            'status' => 'required'
```
Comment (SankalanSaha) [inline app/Http/Controllers/Api/ProjectController.php:239]: Is the status field required?

File: config/mimes.php [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+return [
+    'doc'  => array('application/msword', 'application/vnd.ms-office'),
+    'docx' => array('application/vnd.openxmlformats-officedocument.wordprocessingml.document', 'application/zip'),
```
Comment (SankalanSaha) [inline config/mimes.php:5]: application/zip is not a docx file
Also add the valid mimetypes for the other file extensions

**Review by SankalanSaha — CHANGES_REQUESTED** [review]

File: config/mimes.php (hunk 4 above)
Comment (DipanwitaDey20) [reply config/mimes.php:5]: deleted the file.

File: app/Helpers/helpers.php
[the remark this answers is not in the document]
Comment (DipanwitaDey20) [reply app/Helpers/helpers.php:1]: resolved

File: app/Http/Controllers/Api/ProjectController.php [hunk 5]
```diff
@@ -29,13 +33,17 @@ public function show(Request $request, Project $project){
...
             'surname' => 'required|string',
             'project_name' => 'required|string',
             'description' => 'required|string',
             'contact_mobile_no' => 'required|string',
             'contact_email' => 'required|string',
+            'images' => 'required|array',
+            'images.*' => 'required|max:'.(config('const.dropzone_max_file_size')*1024).'|mimetypes:'.config('const.dropzone_accepted_image'),
+        ],$messages = [
+            'mimes' => 'Please insert image only',
+            'max'   => 'Image should be less than 5 MB'
```
[the remark this answers is not in the document]
Comment (DipanwitaDey20) [reply app/Http/Controllers/Api/ProjectController.php:46]: resolved

File: app/Http/Controllers/Api/ProjectController.php (hunk 1 above)
Comment (DipanwitaDey20) [reply app/Http/Controllers/Api/ProjectController.php:45]: resolved

File: app/Http/Controllers/Api/ProjectController.php [hunk 6]
```diff
@@ -64,68 +106,145 @@ public function add_project(Request $request){
...
+                );
+            }
             return response()->json(['message'=>"Project updated successfully"], 200);
         } catch(Exception $e){
             return response()->json($e->getMessage(), 500);
         }
     }
 
-    public function get_file(Request $request){
+    // public function get_file(Request $request){
```
[the remark this answers is not in the document]
Comment (DipanwitaDey20) [reply app/Http/Controllers/Api/ProjectController.php:174]: resolved

File: app/Http/Controllers/Api/ProjectController.php (hunk 3 above)
Comment (DipanwitaDey20) [reply app/Http/Controllers/Api/ProjectController.php:239]: yes

File: app/Http/Controllers/Api/ProjectController.php [hunk 7]
```diff
@@ -44,18 +52,52 @@ public function add_project(Request $request){
...
+                'status' => $request->status,
+                'reviewer_id' => null,
+                'reviewer_status' => null,
+                'reviewer_status_updated_at' => null,
                 'notes'=> $request->notes
             ]);
             if (!$result) {
                 throw ValidationException::withMessages(['message' => 'Something went wrong, please try again!'], 400);
             }
+            if ($result) {
```
[the remark this answers is not in the document]
Comment (DipanwitaDey20) [reply app/Http/Controllers/Api/ProjectController.php:81]: resolved

File: app/Http/Controllers/Api/ProjectController.php [hunk 8]
```diff
@@ -44,18 +52,52 @@ public function add_project(Request $request){
...
+                        'file_original_name' => $fileName,
+                        'file_extension' => $extension,
+                        'url' => $path
+                    ]);
+                }
+            }
+
             return response()->json(['message'=>'Project saved successfully'],200);
         } catch(Exception $e){
             return response()->json($e->getMessage(),500);
```
[the remark this answers is not in the document]
Comment (DipanwitaDey20) [reply app/Http/Controllers/Api/ProjectController.php:103]: resolved

File: app/Http/Controllers/Api/ProjectController.php [hunk 9]
```diff
@@ -29,13 +33,17 @@ public function show(Request $request, Project $project){
 
     public function add_project(Request $request){
         $validator = Validator::make($request->all(), [
```
[the remark this answers is not in the document]
Comment (DipanwitaDey20) [reply app/Http/Controllers/Api/ProjectController.php:35]: resolved
